Source/Core/Common/FixedSizeQueue.h

```diff
@@ -73,15 +112,58 @@ class FixedSizeQueue
     return temp;
   }
 
+  // From front
+  void erase(size_t num)
+  {
+    assert(count >= num);
+    if constexpr (!std::is_trivial_v<T>)
+    {
+      while (num > 0)
+      {
+        storage[head] = {};
+        head = (head + 1) % N;
+        count--;
+        num--;
+      }
+    }
+    else
+    {
+      head = (head + num) % N;
+      count -= num;
+    }
+  }
+
+  // From head/front, not absolute index
+  T& operator[](size_t index)
+  {
+    assert(index < count);
+    return storage[(head + index) % N];
+  }
+  const T& operator[](size_t index) const
+  {
+    assert(index < count);
+    return storage[(head + index) % N];
+  }
+
   T& front() noexcept { return storage[head]; }
   const T& front() const noexcept { return storage[head]; }
+  T& back() noexcept { return storage[tail]; }
+  const T& back() const noexcept { return storage[tail]; }
+  // Returns the first index of the internal array. Only use if head_to_end() > size()
```

Comment:
Do we really want this to be part of the public API? Seems easy to misuse and I can't think of any situation where external code would need to use this.

Reply:
I think I made these before adding the push_array and copy_to_array methods, as I was doing the same thing but in some external code. They aren't strictly needed sure, but I think they would be nice to have if someone wanted to do some direct memory copying which isn't natively implemented.